esphome/components/micronova/sensor/__init__.py

```diff
@@ -0,0 +1,134 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor
+from esphome.const import (
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_CELSIUS,
+)
+
+from .. import (
+    MicroNova,
+    MicroNovaFunctions,
+    CONF_MICRONOVA_ID,
+    CONF_MEMORY_LOCATION,
+    CONF_MEMORY_ADDRESS,
+    MICRONOVA_LISTENER_SCHEMA,
+    micronova_ns,
+)
+
+UNIT_RPM = "rpm"
+
+MicroNovaSensor = micronova_ns.class_("MicroNovaSensor", sensor.Sensor, cg.Component)
+
+CONF_ROOM_TEMPERATURE = "room_temperature"
+CONF_THERMOSTAT_TEMPERATURE = "thermostat_temperature"
+CONF_FUMES_TEMPERATURE = "fumes_temperature"
+CONF_STOVE_POWER = "stove_power"
+CONF_FAN_SPEED = "fan_speed"
+CONF_MEMORY_ADDRESS_SENSOR = "memory_address_sensor"
+CONF_FAN_RPM_OFFSET = "fan_rpm_offset"
+
+TYPES = [
+    CONF_ROOM_TEMPERATURE,
+    CONF_THERMOSTAT_TEMPERATURE,
+    CONF_FUMES_TEMPERATURE,
+    CONF_STOVE_POWER,
+    CONF_FAN_SPEED,
+    CONF_MEMORY_ADDRESS_SENSOR,
+    CONF_FAN_RPM_OFFSET,
+]
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(CONF_MICRONOVA_ID): cv.use_id(MicroNova),
+        cv.Optional(CONF_ROOM_TEMPERATURE): sensor.sensor_schema(
+            MicroNovaSensor,
+            unit_of_measurement=UNIT_CELSIUS,
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            state_class=STATE_CLASS_MEASUREMENT,
+            accuracy_decimals=1,
+        ).extend(MICRONOVA_LISTENER_SCHEMA),
+        cv.Optional(CONF_THERMOSTAT_TEMPERATURE): sensor.sensor_schema(
+            MicroNovaSensor,
+            unit_of_measurement=UNIT_CELSIUS,
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            state_class=STATE_CLASS_MEASUREMENT,
+            accuracy_decimals=1,
+        ).extend(MICRONOVA_LISTENER_SCHEMA),
+        cv.Optional(CONF_FUMES_TEMPERATURE): sensor.sensor_schema(
+            MicroNovaSensor,
+            unit_of_measurement=UNIT_CELSIUS,
+            device_class=DEVICE_CLASS_TEMPERATURE,
+            state_class=STATE_CLASS_MEASUREMENT,
+            accuracy_decimals=1,
+        ).extend(MICRONOVA_LISTENER_SCHEMA),
+        cv.Optional(CONF_STOVE_POWER): sensor.sensor_schema(
+            MicroNovaSensor,
+            state_class=STATE_CLASS_MEASUREMENT,
+            accuracy_decimals=0,
+        ).extend(MICRONOVA_LISTENER_SCHEMA),
+        cv.Optional(CONF_FAN_SPEED): sensor.sensor_schema(
+            MicroNovaSensor,
+            state_class=STATE_CLASS_MEASUREMENT,
+            unit_of_measurement=UNIT_RPM,
```

Comment:
This already exists in const.py
```suggestion
            unit_of_measurement=UNIT_REVOLUTIONS_PER_MINUTE,
```